### ISSUE TITLE
Override User-Agent?

### ISSUE THREAD

**Issue by Dan203** [mseminatore/TeslaJS#149]
Is there a way to override the built in User-Agent header sent via the http requests? Short of manually editing the .js file?

It appears that the User-Agent is important for the auth functions, like login and refresh, and I'd like to set a value that's more specific to my app rather than the ones pulled from the android app you appear to be using. I've tested with PostMan and it doesn't seem to care what the value is, but it has to be the same for both the 'password' and 'refresh_token' commands or you get an error.

**Comment by mseminatore** [comment]
We could expose a method allowing the UA to be get/set. Is that something you would want to tackle?

**Comment by mseminatore** [comment]
Do you need to set just the **user-agent** or both the **user-agent** and the **x-tesla-user-agent**

**Comment by Dan203** [comment]
Something else is going on here, that I don't think is linked to the user-agent like I thought. I'm just learning NodeJS and all I'm trying to do right now is set up a Firebase project where when the calls a function it pulls the token from the DB and uses it, but if it's expired I want to automatically generate a new token using the refresh function. I set up a DB record using the data I generated in PostMan using my own credentials and then changed the expiration so it would trigger the refresh. When I called the refresh command I get the following error...

"The provided authorization grant is invalid, expired, revoked, does not match the redirection URI used in the authorization request, or was issued to another client."

I thought that was because of the user-agent, but I was just playing with PostMan and changed the user-agent for the initial call and the refresh call and it still worked fine. 

Now I'm not sure what is causing this. Maybe the IP address? The PostMan call is coming from my local machine while the refresh call is coming from Firebase, so I guess that could be it. 

I'm still new to all this (Firebase and NodeJS) so I could be doing something wrong.

Ignore this for now. It may not be as important as I thought. I'm going to have to play with it more and try to figure out what's going on.
